### ISSUE TITLE
Using unusual date formats cause strange datepicker behavior

### ISSUE THREAD

**Issue by stooczu** [themesberg/flowbite-datepicker#40]
Using wierd formats like `DDD dd MM yyyy` or `DD • dd MM yyyy` cause datepicker to select a date different to what user has selected when unfocused.

![image](https://github.com/themesberg/flowbite-datepicker/assets/14873363/c9e00ca8-050b-485e-a951-a88092539182)

on clicked outside:

![image](https://github.com/themesberg/flowbite-datepicker/assets/14873363/c9b5b2f8-7dc2-4c0b-a214-347fd7adde26)

Seems like unfocusing the component triggers processing datepicker value. When you click on the calendar cell, the value it holds is epoch which lets datepicker to process and format the value correctly. When unfocused, the value beeing processed is no longer the epoch value, but it processes the `displayValue` - in this case it's `Thursday • 10 October 2024`. Datepicker starts to process displayValue instead of value which causes it to bug.

Funny thing is the vanilla JS Date class is handles this wierd formatting correctly.
`>new Date('Thursday • 10 October 2024')`
`Thu Oct 10 2024 00:00:00 GMT+0200`

**Comment by sjwatts119** [comment]
Also experiencing this issue

**Comment by bocharsky-bw** [comment]
Yeah, the same for me. My format date: `M d, yyyy` that should render `Mar 31, 2025`. It renders the correct format, but on unfocus, it changes the date to a weird - the year jumps to 2030  for me. :/
